File: stubs/java/src/main/java/com/beastmode/rmddd/examples/ecommerce/OrderCancelledEvent.java
package com.beastmode.rmddd.examples.ecommerce;

import com.beastmode.rmddd.domain.DomainEvent;

import java.util.Map;

/**
 * Domain event fired when an order is cancelled.
 */
public class OrderCancelledEvent extends DomainEvent {
    
    public OrderCancelledEvent(OrderId orderId) {
        super(orderId);
    }
    
    public OrderId getOrderId() {
        return (OrderId) getAggregateId();
    }
    
    @Override
    public Map<String, Object> getEventData() {
        return Map.of(
            "order_id", getOrderId().getValue().toString()
        );
    }
}

File: stubs/csharp/src/BeastMode.RmDdd.Examples/Examples/ECommerce/Events/OrderCancelledEvent.cs
using System.Collections.Generic;
using BeastMode.RmDdd.Domain;

namespace BeastMode.RmDdd.Examples.ECommerce.Events
{
    /// <summary>
    /// Domain event fired when an order is cancelled.
    /// </summary>
    public class OrderCancelledEvent : DomainEvent
    {
        public OrderCancelledEvent(OrderId orderId) : base(orderId)
        {
        }
        
        public OrderId OrderId => (OrderId)AggregateId;
        
        public override IDictionary<string, object> GetEventData()
        {
            return new Dictionary<string, object>
            {
                ["order_id"] = OrderId.Value.ToString()
            };
        }
    }
}